bundle/config/mutator/resolve_variable_references.go

```diff
@@ -125,6 +126,19 @@ func (m *resolveVariableReferences) Apply(ctx context.Context, b *bundle.Bundle)
 	varPath := dyn.NewPath(dyn.Key("var"))
 
 	var diags diag.Diagnostics
+
+	if config.IsExplicitlyEnabled(b.Config.Presets.SourceLinkedDeployment) {
+		revertWorkspacePath, err := updateWorkspacePathForSourceLinkedDeployment(b)
+		if err != nil {
+			diags = diags.Extend(diag.FromErr(err))
+		} else {
+			defer func() {
+				err := revertWorkspacePath()
+				diags = diags.Extend(diag.FromErr(err))
```

Comment:
Can you confirm this actually works (extending diags in defer call?). I guess it should since diags is a slice but not 100% sure